controller/server/controller.go

```diff
@@ -349,34 +350,43 @@ func (c *Controller) RunAndWait() {
 	c.waitForShutdown()
 }
 
-func (c *Controller) waitForShutdown() {
-	log := pfxlog.Logger()
-	ch := make(chan os.Signal, 1)
-	signal.Notify(ch, os.Interrupt)
+func (c *Controller) shutdown(ctx context.Context) {
 
-	<-ch
+}
 
-	if c.config.Enabled {
-		ctx, cancel := context.WithTimeout(context.Background(), time.Second*15)
-		defer cancel()
+func (c *Controller) waitForShutdown() {
 
-		c.apiServer.Shutdown(ctx)
-	}
+	ch := make(chan os.Signal, 1)
+	signal.Notify(ch, os.Interrupt, syscall.SIGTERM)
 
-	log.Info("shutting down")
-	os.Exit(0)
+	<-ch
+	c.Shutdown()
 }
 
 func (c *Controller) Shutdown() {
+	log := pfxlog.Logger()
+
 	ctx, cancel := context.WithTimeout(context.Background(), time.Second*15)
 	defer cancel()
 
+	pfxlog.Logger().Info("edge controller: shutting down...")
 	c.apiServer.Shutdown(ctx)
-	// _ = c.policyEngine.Stop()
 
 	c.AppEnv.Broker.Stop()
 
-	pfxlog.Logger().Info("edge controller shutting down")
+	c.AppEnv.GetHandlers().ApiSession.HeartbeatCollector.Stop()
+
+	pfxlog.Logger().Info("edge controller: stopped")
+
+	pfxlog.Logger().Info("fabric controller: shutting down...")
+
+	c.AppEnv.GetHostController().Shutdown()
```

Comment:
This shuts down the fabric controller gracefully. For some time we haven't been and that the process running w/o calling os.Exit().